### PR TITLE
Handled error in list deployment's domains

### PR DESCRIPTION
**Pull request by MohamedElmdary**
### Description
- Handled error in list deployment's domains

### Changes
- Add try/catch block to show error in `weblet layout` whenever we fail to list
- Remove disconnecting from gridclient from `onMount`
- Add checking for prefix to match omar's pr
- add global timeout
- update tooltip to match omar's pr (add and deployment name) to prefix
- rename gateway to domain

### Related Issues
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1531
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1518

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by amiraabouhadid — COMMENTED** [review]
image
doesn't timeout with network offline

**Comment by MohamedElmdary** [comment]
Tried to deploy with offline network but it timeouts 
image

**Review by 0oM4R — CHANGES_REQUESTED** [review]

File: packages/playground/src/components/manage_gateway_dialog.vue [hunk 1]
```diff
@@ -23,6 +23,38 @@
...
+        </v-alert>
+
+        <v-dialog v-model="failedDomainDialog" max-width="400px">
+          <v-card>
+            <v-card-title class="bg-warning">Failed Domains</v-card-title>
+
+            <v-card-text>
+              <v-list lines="one">
+                <v-list-item v-for="gw in failedToListGws" :key="gw">
+                  <span>* {{ gw }}</span>
```
[the remark this answers is not in the document]
Comment (MohamedElmdary) [reply packages/playground/src/components/manage_gateway_dialog.vue:51]: We already using v-list

but I just notice we can omit(fvm800)

[the remark this answers is not in the document]
Comment (0oM4R) [reply packages/playground/src/components/manage_gateway_dialog.vue:51]: I'm not sure if the asterisk is the best choice here

[the remark this answers is not in the document]
Comment (MohamedElmdary) [reply packages/playground/src/components/manage_gateway_dialog.vue:51]: image

Updated

**Review by amiraabouhadid — CHANGES_REQUESTED** [review]
works 
image
but please use global timeout

File: packages/playground/src/components/manage_gateway_dialog.vue [hunk 2]
```diff
@@ -41,16 +41,16 @@
...
 
             <v-card-text>
-              <v-list lines="one">
-                <v-list-item v-for="gw in failedToListGws" :key="gw">
-                  <span>* {{ gw }}</span>
-                </v-list-item>
-              </v-list>
+              <ul style="list-style: square">
+                <li v-for="gw in failedToListGws" :key="gw">
+                  <span>{{ gw.slice(prefix.length) }}</span>
```
Comment (0oM4R) [inline packages/playground/src/components/manage_gateway_dialog.vue:51]: image

its better to leave it as it is, we have to handle the old gateways or any prefix changes

**Review by 0oM4R — CHANGES_REQUESTED** [review]

**Review by amiraabouhadid — APPROVED** [review]

File: packages/playground/src/components/manage_gateway_dialog.vue [hunk 3]
```diff
@@ -67,7 +113,7 @@
         <div v-show="gatewayTab === 1">
           <form-validator v-model="valid">
             <input-tooltip
-              :tooltip="`Selecting custom domain sets subdomain as gateway name. Prefix(${prefix}) is project name and twin ID.`"
+              :tooltip="`Selecting custom domain sets subdomain as gateway name. Prefix(${prefix}) is solution name, twin ID and deployment name.`"
```
Comment (AhmedHanafy725) [inline packages/playground/src/components/manage_gateway_dialog.vue:116]:
```suggestion
              :tooltip="`Selecting custom domain sets subdomain as gateway name. Prefix(${prefix}) is solution name, twin ID, and deployment name.`"
```

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
select a gateway -> select a domain
image

**Review by 0oM4R — APPROVED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**